include/aos/common/thread.hpp

```diff
@@ -375,6 +386,158 @@ class ConditionalVariable : private NonCopyable {
     Error          mError;
 };
 
+/**
+ * Aos thread pool.
+ *
+ * @tparam cNumThreads number of threads used to perform tasks.
+ * @tparam cQueueSize tasks queue size.
+ * @tparam cMaxTaskSize max task size.
+ */
+template <size_t cNumThreads = 1, size_t cQueueSize = cDefaultThreadPoolQueueSize,
+    size_t cMaxTaskSize = cDefaultThreadPoolMaxTaskSize>
+class ThreadPool : private NonCopyable {
+public:
+    /**
+     * Creates thread pool instance.
+     */
+    ThreadPool()
+        : mCondVar(mMutex)
+        , mShutdown(false)
+        , mThreadsCount(0)
+    {
+    }
+
+    /**
+     * Adds task to task queue.
+     *
+     * @tparam T task type.
+     * @param functor task functor.
+     * @param arg argument passed to task functor when executed.
+     * @return Error.
+     */
+    template <typename T>
+    Error AddTask(T functor, void* arg = nullptr)
+    {
+        static_assert(sizeof(Function<T>) < cMaxTaskSize, "task is bigger than max task size");
+
+        LockGuard lock(mMutex);
+
+        auto err = mBuffer.PushValue(Function<T>(functor, arg));
+        if (!err.IsNone()) {
+            return err;
+        }
+
+        if (!(err = mCondVar.NotifyAll()).IsNone()) {
```

Comment:
You can use NotifyOne to wake up only one thread instead of all available

Reply:
I can not, because it could be one waiting in `Wait` api.